File: get_chat_id.py
import requests
import time
import os
from dotenv import load_dotenv

load_dotenv()

# Telegram Bot Token (.env)
TELEGRAM_BOT_TOKEN = os.getenv("TELEGRAM_BOT_TOKEN", "")

def get_chat_id():
    """Get the Chat ID of a user who sends a message to your bot."""
    if not TELEGRAM_BOT_TOKEN:
        print("❌ TELEGRAM_BOT_TOKEN is missing in .env")
        return None

    print("=" * 60)
    print("📱 TELEGRAM CHAT ID FINDER")
    print("=" * 60)
    print("\n1. Open your bot in Telegram")
    print("2. Send any message to the bot (e.g. 'Hello' or '/start')")
    print("3. After sending, this program will automatically show your Chat ID")
    print("\n⏳ Waiting for a message... (Ctrl+C to exit)\n")
    
    last_update_id = 0
    
    try:
        while True:
            # Get updates from bot
            url = f"https://api.telegram.org/bot{TELEGRAM_BOT_TOKEN}/getUpdates"
            params = {"offset": last_update_id + 1, "timeout": 30}
            
            response = requests.get(url, params=params)
            
            if response.status_code == 200:
                data = response.json()
                
                if data["ok"] and data["result"]:
                    for update in data["result"]:
                        last_update_id = update["update_id"]
                        
                        # If there is a message
                        if "message" in update:
                            chat_id = update["message"]["chat"]["id"]
                            user = update["message"]["from"]
                            username = user.get("username", "N/A")
                            first_name = user.get("first_name", "")
                            last_name = user.get("last_name", "")
                            
                            print("=" * 60)
                            print("✅ CHAT ID FOUND!")
                            print("=" * 60)
                            print(f"\n📋 Chat ID: {chat_id}")
                            print(f"👤 Name: {first_name} {last_name}".strip())
                            print(f"🔗 Username: @{username}")
                            print("\n" + "=" * 60)
                            print("Add this Chat ID to your .env file:")
                            print(f'TELEGRAM_CHAT_ID = "{chat_id}"')
                            print("=" * 60)
                            
                            # Send chat ID back to user
                            send_url = f"https://api.telegram.org/bot{TELEGRAM_BOT_TOKEN}/sendMessage"
                            message = f"✅ Connection successful!\n\n📋 Your Chat ID: <code>{chat_id}</code>\n\nYou can add this ID to your code."
                            payload = {
                                "chat_id": chat_id,
                                "text": message,
                                "parse_mode": "HTML"
                            }
                            requests.post(send_url, json=payload)
                            
                            return chat_id
            else:
                print(f"❌ Error: {response.status_code}")
            
            time.sleep(1)
            
    except KeyboardInterrupt:
        print("\n\n⛔ Program stopped")
        return None
    except Exception as e:
        print(f"\n❌ An error occurred: {e}")
        return None

if __name__ == "__main__":
    get_chat_id()

File: get_chat_id.js
const axios = require('axios');
require('dotenv').config();

// Telegram Bot Token (.env)
const TELEGRAM_BOT_TOKEN = process.env.TELEGRAM_BOT_TOKEN || '';

async function getChatId() {
    /**
     * Get the Chat ID of a user who sends a message to your bot.
     */
    if (!TELEGRAM_BOT_TOKEN) {
        console.log('❌ TELEGRAM_BOT_TOKEN is missing in .env');
        return null;
    }

    console.log('='.repeat(60));
    console.log('📱 TELEGRAM CHAT ID FINDER');
    console.log('='.repeat(60));
    console.log('\n1. Open your bot in Telegram');
    console.log('2. Send any message to the bot (e.g. \'Hello\' or \'/start\')');
    console.log('3. After sending, this program will automatically show your Chat ID');
    console.log('\n⏳ Waiting for a message... (Ctrl+C to exit)\n');
    
    let lastUpdateId = 0;
    
    try {
        while (true) {
            // Get updates from bot
            const url = `https://api.telegram.org/bot${TELEGRAM_BOT_TOKEN}/getUpdates`;
            const params = {
                offset: lastUpdateId + 1,
                timeout: 30
            };
            
            const response = await axios.get(url, { params });
            
            if (response.status === 200) {
                const data = response.data;
                
                if (data.ok && data.result && data.result.length > 0) {
                    for (const update of data.result) {
                        lastUpdateId = update.update_id;
                        
                        // If there is a message
                        if (update.message) {
                            const chatId = update.message.chat.id;
                            const user = update.message.from;
                            const username = user.username || 'N/A';
                            const firstName = user.first_name || '';
                            const lastName = user.last_name || '';
                            
                            console.log('='.repeat(60));
                            console.log('✅ CHAT ID FOUND!');
                            console.log('='.repeat(60));
                            console.log(`\n📋 Chat ID: ${chatId}`);
                            console.log(`👤 Name: ${firstName} ${lastName}`.trim());
                            console.log(`🔗 Username: @${username}`);
                            console.log('\n' + '='.repeat(60));
                            console.log('Add this Chat ID to your .env file:');
                            console.log(`TELEGRAM_CHAT_ID="${chatId}"`);
                            console.log('='.repeat(60));
                            
                            // Send chat ID back to user
                            const sendUrl = `https://api.telegram.org/bot${TELEGRAM_BOT_TOKEN}/sendMessage`;
                            const message = `✅ Connection successful!\n\n📋 Your Chat ID: <code>${chatId}</code>\n\nYou can add this ID to your code.`;
                            const payload = {
                                chat_id: chatId,
                                text: message,
                                parse_mode: 'HTML'
                            };
                            await axios.post(sendUrl, payload);
                            
                            return chatId;
                        }
                    }
                }
            } else {
                console.log(`❌ Error: ${response.status}`);
            }
            
            await new Promise(resolve => setTimeout(resolve, 1000));
        }
    } catch (error) {
        if (error.message === 'SIGINT') {
            console.log('\n\n⛔ Program stopped');
        } else {
            console.log(`\n❌ An error occurred: ${error.message}`);
        }
        return null;
    }
}

// Handle Ctrl+C
process.on('SIGINT', () => {
    console.log('\n\n⛔ Program stopped');
    process.exit(0);
});

if (require.main === module) {
    getChatId();
}

module.exports = { getChatId };
